R/utils.R

```diff
@@ -69,11 +69,18 @@ MakeMS2CosineDataframe <- function(df) {
 #' Intensity is scaled to 100 and filtered to drop all intensity values below 0.5.
 #'
 MakeScantable <- function(scan) {
+  # Variable names here could also be improved - this converts an MS2 string to an MS2 data frame, right? "Scan" is not super informative
   requireNamespace("dplyr", quietly = TRUE)
+  # Not sure I love this method of converting string to df - something like
+  # scan %>% data.frame() %>%
+  # separate_rows(scan, sep = "; ") %>%
+  # separate(v, into = c("mz", "int"), sep = ", ")
+  # is a little more elegant and easier to read
   scantable <- read.table(text = as.character(scan),
                           col.names = c("mz", "intensity"), fill = TRUE) %>%
     dplyr::mutate(mz = as.numeric(mz %>% stringr::str_replace(",", "")),
                   intensity = as.numeric(intensity %>% stringr::str_replace(";", "")),
+                  # Isn't intensity already scaled? Doesn't need to be rescaled here
```

Comment:
I want to leave that in there in case someone makes a mistake and forgets to scale it. Alternatively that could be one of the checks at the beginning, but I feel that leaving it in doesn't matter.